.github/workflows/build-rust-cross-platform.yml

```
@@ -0,0 +1,63 @@
+name: Build Rust Cross Platform
+
+on:
+  workflow_call:
+
+jobs:
+  build_rust:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        arch: [x64, arm64]
+        exclude:
+          - os: windows-latest
+            arch: arm64
+          - os: ubuntu-latest
+            arch: arm64
```

I would prefer if we follow the `build-cli.yml` structure here. It should make it more maintainable long term and simplify adding more architectures. This also lets you delete "Set build architecture"

```suggestion
        settings:
          - os: macos-12
            target: x86_64-apple-darwin

          - os: macos-12
            target: aarch64-apple-darwin

          - os: windows-2022
            target: x86_64-pc-windows-msvc

          - os: windows-2022
            target: aarch64-pc-windows-msvc

          - os: ubuntu-22.04
            target: x86_64-unknown-linux-gnu

          - os: ubuntu-22.04
            target: aarch64-unknown-linux-gnu
```

---

Changed